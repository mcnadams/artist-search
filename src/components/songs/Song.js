import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Song({ song, artist }) {
  const artistURI = encodeURIComponent(artist);
  const songURI = encodeURIComponent(song.title);
  const path = `/song/${artistURI}/${songURI}`;
  return (
    <>
      <Link to={path} style={{ color: 'black', padding: '5px', fontSize: '1em' }}>
        {song.title}
      </Link>
    </>
  );
}

Song.propTypes = {
  song: PropTypes.object.isRequired,
  artist: PropTypes.string.isRequired
};

export default Song;
