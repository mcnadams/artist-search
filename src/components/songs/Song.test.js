import React from 'react';
import { shallow } from 'enzyme';
import Song from './Song';

describe('Song renders', () => {
  it('renders Song', () => {
    const wrapper = shallow(<Song artist={'lady'} song={{ date: 'yes', id: 'no' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
