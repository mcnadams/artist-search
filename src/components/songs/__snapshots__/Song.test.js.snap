// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song renders renders Song 1`] = `
<Fragment>
  <Link
    style={
      Object {
        "color": "black",
        "fontSize": "1em",
        "padding": "5px",
      }
    }
    to="/song/lady/undefined"
  />
</Fragment>
`;
